### PR TITLE
[VEN-1909]: Add VTreasuryV8

### PR DESCRIPTION
**Pull request by GitGuru7**
## Description

Adds new VTreasuryV8 contract in 0.8.20 solidity version with deployment script

Resolves VEN-1909

## Checklist

- [ ] I have updated the documentation to account for the changes in the code.
- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
- [ ] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).

### CODE REVIEW

File: contracts/Governance/VTreasuryV8.sol [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+ * @notice Protocol treasury that holds tokens owned by Venus
+ */
+contract VTreasuryV8 is Ownable {
+    using SafeERC20 for IERC20;
+
+    // WithdrawTreasuryToken Event
+    event WithdrawTreasuryToken(address tokenAddress, uint256 withdrawAmount, address withdrawAddress);
+
+    // WithdrawTreasuryNative Event
+    event WithdrawTreasuryNative(uint256 withdrawAmount, address withdrawAddress);
```
Comment (chechu) [inline contracts/Governance/VTreasuryV8.sol:18]: We could index the addresses of the events

**Review by chechu — APPROVED** [review]

File: contracts/Governance/VTreasuryV8.sol (hunk 1 above)
Comment (GitGuru7) [reply contracts/Governance/VTreasuryV8.sol]: Added here [29b65b8](https://github.com/VenusProtocol/venus-protocol/pull/345/commits/29b65b8e0767eeef1f45af9c292be6b750c31282)

**Review by chechu — APPROVED** [review]

File: contracts/Governance/VTreasuryV8.sol [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        uint256 actualWithdrawAmount = withdrawAmount;
+        // Get Treasury Native Balance
+        uint256 nativeBalance = address(this).balance;
+
+        // Check Withdraw Amount
+        if (withdrawAmount > nativeBalance) {
+            // Update actualWithdrawAmount
+            actualWithdrawAmount = nativeBalance;
+        }
+        // Transfer BNB to withdrawAddress
```
Comment (chechu) [inline contracts/Governance/VTreasuryV8.sol:75]:
```suggestion
        // Transfer the native token to withdrawAddress
```

**Review by Debugger022 — APPROVED** [review]

**Review by Debugger022 — APPROVED** [review]

File: contracts/Governance/VTreasuryV8.sol (hunk 2 above)
Comment (GitGuru7) [reply contracts/Governance/VTreasuryV8.sol]: [acab6c6](https://github.com/VenusProtocol/venus-protocol/pull/345/commits/acab6c6fb0c33387d0081b9797b8efb40127c362)

**Review by chechu — APPROVED** [review]

**Review by chechu — APPROVED** [review]

**Pull request merged**